static/js/main.js

```diff
@@ -1,12 +1,12 @@
 function findCurrentSection(htmlEl, mainEl) {
   const windowHeight = htmlEl.clientHeight;
-  const scrollHeight = htmlEl.scrollHeight;
-  const scrollTop = htmlEl.scrollTop;
 
-  // figure out which section we're in. basically, in order to make this work well for the first and last sections, we:
-  // 1. pick a point as far down the window as we are down the scrollbar and in the middle of the main element
-  const scrollPct = scrollTop / (scrollHeight - windowHeight);
-  const targetY = (windowHeight - 1) * scrollPct; // targetY is a position but windowHeight is a length, -1 to avoid fencepost error
+  // targetY is a position but windowHeight is a length, -1 to avoid fencepost error
+  // the footnotes are long enough that scrolling to the bottom would highlight the last section
+  // so I picked a point after the header that is a reasonable place for a reader to be looking at
+  // it's also sized such that a click on a section in the sidebar will result in that section being highlit
+  const targetY = (windowHeight - 1) * .15;
```

Comment:
The `// targetY is a position...` comment is directly applicable to the line - you should move it back so it's on the end of the line.

It took me a bit to figure out what was happening here with the comments (and you've deleted the summary of the algorithm and the `// 1. ...` comment to match the `// 2.` and `// 3.` below). How about something like:

```
// Figure out which section we're in. We have to take into account both the size of the footnote section and
// the position of the page after clicking a link in the sidebar. So we:
// 1. Focus on the point 15% of the way down the window (a reasonable place for the reader to be looking at) and
// in the middle of the main section
const targetY = (windowHeight - 1) * .15; // targetY is a position but windowHeight is a length, so -1 to avoid a fencepost error
const targetX = (mainEl.offsetLeft + mainEl.scrollWidth) / 2;
```

Reply:
I never use end-of-line comments; I always put them above. (I prefer to read the comment before the code.)